src/classes/groundplaces.ts

```diff
@@ -47,13 +54,21 @@ export class GroundPlacesController {
   }
 
   /**
-   * @description Returns a list of places.
+   * @description Returns a list of ground places.
    * @param {string} query - Can be a name, a Gpuid, a unique name or other name.
-   * @param {AutoCompleteFilters} filters - Filters with different options (StopGroup, StopCluster, Serviced, SegmentProvider).
+   * @param {AutoCompleteFilters[]|undefined} filters - Filters with different options (StopGroup, StopCluster, Serviced, SegmentProvider).
+   *
+   * If you do not give filters, the list will not be filtered.
    * @returns {AutoComplete[]}
    */
   // @ts-ignore
-  public autocomplete(query: string, filters: AutoCompleteFilters[]): AutoComplete[] {}
+  public autocomplete(query: string, filters?: AutoCompleteFilters[]): AutoComplete[] {
+    const groundPlacesFiltered: AutoComplete[] = this.storageService.filterGroundPlaces(filters);
+
+    const groundPlaces: AutoComplete[] = this.storageService.searchGroundPlaces(groundPlacesFiltered, query);
```

Comment:
To me the function should only be on the storage service.

Reply:
Simple search('query', {filtrers}

Reply:
Not sure that I understand "To me the function should only be on the storage service.", If you are referring at the `searchGroundPlaces` method, this is actually on the Storage class.

Reply:
And if you suggest to create another method called search that take two parameters like ('query', filters), it's actually what `autocomplete` method does, right?

Reply:
Simplicity is king. 
One autcomplete one function. No need to do 10 functions to do that. 
Go straight to the point.